### PR TITLE
Fixed 0D Fp32 issue and added test

### PR DESCRIPTION
**Pull request by GopikrishnaGitRepo**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [x] bug
- [ ] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

Fixes an SDSC generation failure when a dtype-conversion path involves a 0-dimensional (scalar) tensor.

For scalar tensors, parse_op_spec() may derive no valid stick dimension, leaving op_stick_dim as None. Subsequent SDSC iteration-space and dimension-order construction assumes at least one dimension is present and can fail with errors such as:

    error: Specified dimension of parametric loop operation is marked to be dropped

This occurs in cases such as x + s,
where x is an FP16 tensor and s is a 0-D FP32 scalar tensor that flows through a dtype-conversion operation.

The fix detects the absence of a stick dimension and injects a virtual stick symbol whose extent is derived from elems_per_stick(). The virtual dimension is added to the iteration space, work_slices, dim_splits, and dimension order so that downstream SDSC generation can proceed uniformly for both scalar and non-scalar tensors. 

The original manifestation of this class of issue was observed in PR #2411, The intial fix for this issue was added in PR #3158

Adds a regression test "test_add_0dim_fp32_scalar_to_fp16_tensor" in "tests/inductor/test_inductor_dtype_scalars.py"

#### Which issue(s) this PR is related to:
#2845

### CODE REVIEW

File: torch_spyre/_inductor/codegen/superdsc.py [hunk 1]
```diff
@@ -755,9 +755,26 @@ def parse_op_spec(op_spec: OpSpec) -> tuple["SDSCSpec", "dict"]:
...
+            0
+        ].device_dtype.elems_per_stick()
+        output_stick_sym = Symbol(OUTPUT_DIM_LABELS[0])
+        sdsc_iteration_space[output_stick_sym] = max(
+            arg.device_dtype.elems_per_stick() for arg in op_spec.args
+        )
+        work_slices[output_stick_sym] = 1
+        dim_splits[output_stick_sym] = 1
+        op_dim_order = [output_stick_sym]
+        op_stick_dim = output_stick_sym
```
Comment (Vikram123-byte) [inline torch_spyre/_inductor/codegen/superdsc.py:1021]: Keep virtual stick and mb symbols distinct.

For the 0-D path, input_stick_sym is Symbol(INPUT_DIM_LABELS[0]), which is the exact symbol later assigned to mb_sym. Consequently, Line 782’s {mb_sym: 1, **sdsc_iteration_space} overwrites the documented virtual mb=1 size with the input dtype’s elems_per_stick(). This can produce an incorrect SDSC iteration space for the regression case.

Also initialize dim_splits and work_slices for every newly inserted input-stick key; otherwise nonzero-rank cases can fail when the later mapping indexes every sdsc_iteration_space dimension.

File: tests/inductor/test_inductor_dtype_scalars.py [hunk 2]
```diff
@@ -373,6 +373,25 @@ def large_scalar_mul(x):
...
+        the SDSC spec builder indexed into empty dim lists.
+        """
+
+        def add_scalar(x, s):
+            return x + s
+
+        x = torch.ones(64, dtype=torch.float16)
+        s = torch.tensor(2.0, dtype=torch.float32)
+        _compare_modes(execution_mode, add_scalar, x, s, atol=1e-3, rtol=1e-3)
+
```
Comment (manid2) [inline tests/inductor/test_inductor_dtype_scalars.py:394]: Please add test to verify type conversion for 0-dim scalars.

**Comment by avery-blanchard** [comment]
@GopikrishnaGitRepo hi! Could you fix the signatures on these commits: [8281acf](https://github.com/torch-spyre/torch-spyre/pull/3348/commits/8281acf01b47e94df0d0d516c54a96f6e0bcf834), [efdf521](https://github.com/torch-spyre/torch-spyre/pull/3348/commits/efdf521a4d79906959c9b2ad8fe8a846cd50b258), [66eb865](https://github.com/torch-spyre/torch-spyre/pull/3348/commits/66eb8651f5c3102b12e2922972306af6e4ee9b28)?

**Comment by GopikrishnaGitRepo** [comment]
@avery-blanchard i have changed the commit signatures and signed it properly. Please verify.

**Comment by avery-blanchard** [comment]
@GopikrishnaGitRepo it looks like a bad rebase maybe (100+ commits and 20 contributors on the PR)? It looks like the DCO is also confused by this.

**Comment by GopikrishnaGitRepo** [comment]
On the latest main branch,
``` error: Specified dimension of parametric loop operation is marked to be dropped ```
and the code is running without any error, However, a new issue remains where adding a 0-D FP32 tensor with a 1-D FP16 tensor produces an incorrect output dtype.

The expected behavior is:

Convert the 0-D FP32 tensor to FP16.
Perform the addition between the 0-D FP16 tensor and the 1-D FP16 tensor.
The output should be FP16.

Currently, the output is incorrectly returned as FP32.

This PR #3912 fixes the incorrect output dtype and ensures the operation follows the expected PyTorch type-promotion behavior.
